pennylane/circuit_drawer/circuit_drawer.py

```diff
@@ -117,55 +121,26 @@ def __init__(
         self.full_representation_grid = self.operation_representation_grid.copy()
         self.full_representation_grid.append_grid_by_layers(self.observable_representation_grid)
 
-    def make_wire_conversion_dicts(self, raw_operation_grid, raw_observable_grid):
-        """Prepare the dictionaries used to convert between internal and device wires.
-
-        This conversion is necessary due to the fact that the circuit drawer internally uses
-        ascending wires that have to be matched to the actual wires of the Operations inside
-        the circuit.
+    def extract_active_register(self, raw_operation_grid, raw_observable_grid):
+        """Get the subset of wires in the register that are used in the circuit.
 
         Args:
             raw_operation_grid (Iterable[~.Operator]): The raw grid of operations
             raw_observable_grid (Iterable[~.Operator]): The raw  grid of observables
+
+        Return:
+            Wires: register of active wires
         """
         # pylint: disable=protected-access
         all_operators = list(qml.utils._flatten(raw_operation_grid)) + list(
             qml.utils._flatten(raw_observable_grid)
         )
-        all_wires = [op.wires for op in all_operators if op is not None]
-        circuit_wires = sorted(set(qml.utils._flatten(all_wires)))
-        internal_wires = list(range(len(circuit_wires)))
-
-        self._cicuit_wire_to_internal_wire = dict(zip(circuit_wires, internal_wires))
-        self._internal_wire_to_circuit_wire = dict(zip(internal_wires, circuit_wires))
-
-    def circuit_wires_to_internal_wires(self, wires):
-        """Convert one or multiple device wires to internal wires.
-
-        Args:
-            wires (Union[Iterable[int],int]): One or multiple device wires
-
-        Returns:
-            Union[Iterable[int],int]: The corresponding internal wires
-        """
-        if isinstance(wires, Iterable):
-            return [self._cicuit_wire_to_internal_wire[wire] for wire in wires]
-
-        return self._cicuit_wire_to_internal_wire[wires]
-
-    def internal_wires_to_circuit_wires(self, wires):
-        """Convert one or multiple internal wires to device wires.
-
-        Args:
-            wires (Union[Iterable[int],int]): One or multiple internal wires
-
-        Returns:
-            Union[Iterable[int],int]: The corresponding device wires
-        """
-        if isinstance(wires, Iterable):
-            return [self._internal_wire_to_circuit_wire[wire] for wire in wires]
-
-        return self._internal_wire_to_circuit_wire[wires]
+        all_wires_with_duplicates = [op.wires for op in all_operators if op is not None]
+        # make Wires object containing all used wires
+        all_wires = Wires.all_wires(all_wires_with_duplicates)
+        # shared wires will observe the ordering of the register
+        shared_wires = Wires.shared_wires([self.register, all_wires])
```

Comment:
A sharp observation :) In this case I am using ``shared_wires`` because the function orders the resulting object with respect to the ordering in the first argument of the list (here ``self.register``). This way the circuit is drawn in the way that the register is structured. But you are right, ``shared_wires`` and ``all_wires`` contain the same wires, only perhaps in a different order.